services/wordpress/wordpress-rating.service.js

```diff
@@ -0,0 +1,162 @@
+'use strict'
+
+const WordpressBase = require('./wordpress-base')
+const { starRating, metric } = require('../../lib/text-formatters')
+
+function colorFromRating({ rating }) {
+  if (rating === 0) {
+    return 'red'
+  } else if (rating < 2) {
+    return 'yellow'
+  } else if (rating < 3) {
+    return 'yellowgreen'
+  } else if (rating < 4) {
+    return 'green'
+  } else {
+    return 'brightgreen'
+  }
+}
+
+class WordpressRatingBase extends WordpressBase {
+  static render({ rating, total }) {
+    return {
+      message: `${rating}/5 (${metric(total)})`,
+      color: colorFromRating({ rating }),
+    }
+  }
+
+  static get category() {
+    return 'rating'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'rating' }
+  }
+
+  async handleResponse({ response }) {
+    const total = response.num_ratings
+    const rating = (response.rating / 100) * 5
+    return this.constructor.render({ rating, total })
+  }
+
+  static get url() {
+    return {
+      base: `wordpress/${this.apiType}/rating`,
```

Comment:
Do you want me to change it to `/r` as stars and keep `/rating` as `*/5` rating?

Reply:
Good point. I hadn't focussed that the existing `/r` maps to the new stars badge. How about we make the primary documented URLs `/rating` and `/stars` but support `/r` (as an alias for stars) to avoid breaking for users who are currently using `/r`.

You can use a non-capturing group in the format regex to do this e.g:

```js
static get url() {
  return {
    base: `wordpress/${this.apiType}`,
    format: '(?:stars|r)/(.+)',
    capture: ['slug'],
  }
}
```